### PR TITLE
No newtonsoft json

### LINKED ISSUE #11: Still using NewtonsoftJson, Convert to System.Text.Json
For dotnet 3.x version, try dropping usage of NewtonsoftJson with replacement of System.Text.Json for all those serialzian/deserialization methods.
As seen https://github.com/IdentityManager/IdentityManager2/blob/449d0676ce91aeec9a8c77fcc5853c48fca88847/src/IdentityManager2/Configuration/DependencyInjection/IdentityManagerServiceCollectionExtensions.cs#L20 and https://github.com/IdentityManager/IdentityManager2/blob/449d0676ce91aeec9a8c77fcc5853c48fca88847/src/IdentityManager2/Api/Controllers/PageController.cs#L32

``                Model = JsonConvert.SerializeObject(new``

I could prepare a PR :-)

### PR DESCRIPTION
**Pull request by ChaosEngine**
fixes https://github.com/IdentityManager/IdentityManager2/issues/11

### CODE REVIEW

File: src/IdentityManager2/Api/Controllers/PageController.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public async Task<IActionResult> Index()
             return View("/Areas/IdentityManager/Pages/Index.cshtml", new PageModel
             {
                 PathBase = Request.PathBase,
-                Model = JsonConvert.SerializeObject(new
+                Model = JsonSerializer.Serialize(new PageModelParams
```
Comment (ChaosEngine) [inline src/IdentityManager2/Api/Controllers/PageController.cs:32]: some real class implementation is needed for System.Text.JSon to work, no annonymous types

File: src/IdentityManager2/Api/Models/RoleQueryResultResource.cs [hunk 2]
```diff
@@ -29,10 +29,8 @@ public RoleQueryResultResource(QueryResult<RoleSummary> result, IUrlHelper url,
         }
     }
 
-    public class RoleQueryResultResourceData : QueryResult<RoleSummary>
+    public class RoleQueryResultResourceData : QueryResult<RoleResultResource>
     {
-        public new IEnumerable<RoleResultResource> Items { get; set; }
```
Comment (ChaosEngine) [inline src/IdentityManager2/Api/Models/RoleQueryResultResource.cs:34]: dropping this is needed for System.Text.Json coz it is lost with new overrides :-/

File: src/IdentityManager2/Api/Models/UserQueryResultResource.cs [hunk 3]
```diff
@@ -30,10 +30,8 @@ public UserQueryResultResource(QueryResult<UserSummary> result, IUrlHelper url,
         }
     }
 
-    public class UserQueryResultResourceData : QueryResult<UserSummary>
+    public class UserQueryResultResourceData : QueryResult<UserResultResource>
     {
-        public new IEnumerable<UserResultResource> Items { get; set; }
```
Comment (ChaosEngine) [inline src/IdentityManager2/Api/Models/UserQueryResultResource.cs:35]: dropping this is needed for System.Text.Json 'coz it is lost with new overrides :-/

File: src/IdentityManager2/Assets/Scripts/App/ttIdmUsers.js [hunk 4]
```diff
@@ -85,7 +85,7 @@
                 var props = properties.map(function (item) {
                     return {
                         type: item.meta.type,
-                        value: item.data
+                        value: item.data.toString()
```
Comment (ChaosEngine) [inline src/IdentityManager2/Assets/Scripts/App/ttIdmUsers.js:88]: this is actually needed to allow serialization of values (int, string, what-not) into https://github.com/IdentityManager/IdentityManager2/blob/449d0676ce91aeec9a8c77fcc5853c48fca88847/src/IdentityManager2/Core/PropertyValue.cs#L10

File: src/IdentityManager2/Configuration/DependencyInjection/IdentityManagerServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -16,9 +16,8 @@ public static IIdentityManagerBuilder AddIdentityManager(this IServiceCollection
             var identityManagerOptions = services.BuildServiceProvider().GetRequiredService<IOptions<IdentityManagerOptions>>().Value;
             identityManagerOptions.Validate();
 
-            services.AddControllersWithViews()
-                .AddNewtonsoftJson();
```
Comment (ChaosEngine) [inline src/IdentityManager2/Configuration/DependencyInjection/IdentityManagerServiceCollectionExtensions.cs:20]: not needed anymore

File: src/IdentityManager2/IdentityManager2.csproj [hunk 6]
```diff
@@ -37,8 +37,6 @@
 
   <ItemGroup>
     <FrameworkReference Include="Microsoft.AspNetCore.App" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="3.1.0" />
-    <PackageReference Include="Microsoft.Extensions.FileProviders.Embedded" Version="3.1.0" />
```
Comment (ChaosEngine) [inline src/IdentityManager2/IdentityManager2.csproj:41]: *Microsoft.Extensions.FileProviders.Embedded* this is also provided by *Microsoft.AspNetCore.App*, no need to duplicate

**Comment by scottbrady91** [comment]
Awesome! Thanks for this, I'll take a look at it now

**Comment by scottbrady91** [comment]
Any news on this one?

**Comment by ChaosEngine** [comment]
Waiting for your approval AFAIK :-)

**Comment by scottbrady91** [comment]
There are a couple of comments inline.

**Review by ChaosEngine — COMMENTED** [review]
reviewed, anything else I can do/fix?

**Comment by ChaosEngine** [comment]
What is the state with this PR? Anything else missing?

**Comment by scottbrady91** [comment]
:shipit:

**Pull request merged**